### DISCUSSION TITLE
Trino gets protobuf or avro data from redis

### DISCUSSION THREAD

**Discussion by sssssssansan** [trinodb/trino#17913] [Q&A]
I use python to establish a connection with redis and write the serialized avro data to redis, and set the configuration file and dataschema path according to the format in the official document, but the error dataschema is null is always reported when querying.

![image](https://github.com/trinodb/trino/assets/65671005/47ba27c8-d53d-4a9e-b288-c79fffe170d0)

I checked the source code and found that currently redis seems to support the parsing of avro data according to the specified format, but there is no variable defined to receive dataschema parameters at the corresponding interface, and the corresponding method and interface are defined in the Kafka object file. Does this mean that trino does not yet support getting avro data from redis and automatically parsing it?
The figure below is the corresponding redis configuration setting source code and kafka setting source code
![image](https://github.com/trinodb/trino/assets/65671005/afe3bd40-63e3-463a-bc5f-6819e739f346)
![image](https://github.com/trinodb/trino/assets/65671005/5061d97b-4e93-4694-bdd2-0d1e22b9625c)

**Comment by hashhar** [comment]
Yes, both the KAfka and REdis connectors require you to provide a table description file to be able to map from the Avro bytes to a table-like structure. Just the avro payload itself is not sufficient since it'll only deserialize into an object but how the object is mapped into a SQL table is not defined.
